src/ray/stats/stats_test.cc

```diff
@@ -204,6 +204,24 @@ TEST_F(StatsTest, TestShutdownTakesLongTime) {
   ray::stats::Shutdown();
 }
 
+DEFINE_stats(test_hist, "TestStats", ("method", "method2"), (1.0, 2.0, 3.0, 4.0),
+             ray::stats::HISTOGRAM);
+DEFINE_stats(test_2, "TestStats", ("method", "method2"), (1.0), ray::stats::COUNT,
+             ray::stats::SUM);
+DEFINE_stats(test, "TestStats", ("method"), (1.0), ray::stats::COUNT, ray::stats::SUM);
+DEFINE_stats(test_declare, "TestStats2", ("tag1"), (1.0), ray::stats::COUNT,
+             ray::stats::SUM);
+DECLARE_stats(test_declare);
+
+TEST_F(StatsTest, STAT_DEF) {
+  ray::stats::Shutdown();
+  std::shared_ptr<stats::MetricExporterClient> exporter(
+      new stats::StdoutExporterClient());
+  ray::stats::Init({}, MetricsAgentPort, exporter);
+  STATS_test.Record(1.0);
+  STATS_test_declare.Record(1.0, "Test");
+}
+
```

Comment:
Can you also verify metrics that are added by this new macro will pass other tests? Especially the ones around shutdown path tests.

Reply:
Do you mean to move them above to the top of the file?

Reply:
Oh no. I meant these tests `MultiThreadedInitializationTest, TestShutdownTakesLongTime` passes with stats defined with the new macro (since they are tested with the old code)

Reply:
I think I understand what you mean here, but I feel these two tests don't apply to the new metrics. They are global stats, and they'll be initialized before `main()`, so no thread issues here. And it won't survive from the shutdown.